cellxgene_schema_cli/cellxgene_schema/validate.py

```diff
@@ -578,8 +577,31 @@ def _validate_column_feature_is_filtered(self, column: pd.Series, column_name: s
             )
             return
 
+        if self.adata.raw is None:
+            if column.any():
+                self.errors.append(
+                    "'feature_is_filtered' must be False for all features if 'adata.raw' is not present."
+                )
+            return
+        else:
+            sum_X = compute_column_sums(self.adata.X)
+            sum_raw_X = compute_column_sums(self.adata.raw.X)
+            for i, _ in enumerate(sum_X):
+                is_filtered = column[i]
+                raw_column_sum = sum_raw_X[i]
+                # If adata.X has all 0s for the column, then is_filtered must be True or adata.raw.X should also be all 0s
+                if sum_X[i] == 0:
+                    if is_filtered or raw_column_sum == 0:
```

Comment:
Is it guaranteed that the order of columns in var, raw.X, and X will always match? i recall we've had this concern with other matrix checks, we might have to enforce canonical format first here (unless thats already guaranteed by a previous step?)

Reply:
if we don't guarnatee canonical format earlier, whether the order can be safely assumed to be the same could be a question for lattice ^

Reply:
seems like according to isaac, we can generally assume that `adata.var_names[i]` will match `adata.X[i]`: https://czi-sci.slack.com/archives/C08MF1AJ6F5/p1746127011430139

it seems like it should be correct most of the time, but i updated the error to also include the index of the mismatch